helper-cli/src/main/kotlin/commands/DownloadResultsFromPostgresCommand.kt

```diff
@@ -0,0 +1,189 @@
+import org.ossreviewtoolkit.model.utils.DatabaseUtils
+import org.ossreviewtoolkit.utils.common.expandTilde
+import org.ossreviewtoolkit.utils.core.ORT_CONFIG_FILENAME
+import org.ossreviewtoolkit.utils.core.ortConfigDirectory
+
+class DownloadResultsFromPostgresCommand : CliktCommand(
+    name = "download-results-from-postgres",
+    help = "Download an ORT result from a PostgreSQL database. The symmetric command to ORT's " +
+            " upload-result-to-postgres command."
+) {
+    private val outputDir by option(
+        "--output-dir", "-o",
+        help = "The directory to write the ORT results to."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = false, canBeDir = true, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val tableName by option(
+        "--table-name",
+        help = "The name of the table which holds the ORT results."
+    ).required()
+
+    private val columnName by option(
+        "--column-name",
+        help = "The name of the JSONB column containing the ORT result."
+    ).required()
+
+    private val configFile by option(
+        "--config",
+        help = "The path to the ORT configuration file that configures the scan results storage."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .default(ortConfigDirectory.resolve(ORT_CONFIG_FILENAME))
+
+    private val configArguments by option(
+        "-P",
+        help = "Override a key-value pair in the configuration file. For example: " +
+                "-P ort.scanner.storages.postgres.schema=testSchema"
+    ).associate()
+
+    private val startId by option(
+        "--start-id",
+        help = "The smallest row-id to download."
+    ).convert { it.toInt() }
+        .default(-1)
+
+    override fun run() {
+        if (!outputDir.exists()) outputDir.mkdirs()
+
+        val localStorage = OrtResultStorage(outputDir)
+        val localRowIds = localStorage.getIds()
+
+        openDatabaseConnection().use { connection ->
+            val remoteRowIds = fetchRowIds(connection)
+            val missingRowIds = remoteRowIds - localRowIds
+            println("Requested the download of ${remoteRowIds.size} of which ${missingRowIds.size} are missing.")
+
+            missingRowIds.forEachIndexed { i, rowId ->
+                println("[${i + 1} / ${missingRowIds.size}] downloading row $rowId")
+
+                val duration = measureTime {
+                    val ortResultJson = fetchOrtResult(connection, rowId)
+
+                    if (ortResultJson != null) localStorage.add(rowId, ortResultJson)
+                }
+
+                println("[${i + 1} / ${missingRowIds.size}] took ${duration.toString(DurationUnit.SECONDS)}s.")
```

Comment:
If there is no requirement to always print the duration in seconds you could use the default `toString()` which chooses matching units automatically.